### DISCUSSION TITLE
Q-Menu Styling around the "portal/teleport"?

### DISCUSSION THREAD

**Discussion by jonathancounihan** [quasarframework/quasar#17524] [General - Components / Directives / etc]
On the q-menu docs page, https://quasar.dev/vue-components/menu#usage, it says the following:

> "QMenu will be injected as a direct child of <body> through a Quasar Portal."

What is a quasar portal? Or `Teleport`? I have seen that term used in the docs but I don't understand what it refers to. Is it just referring to moving content around the page?

Also how can I style around the q-menu? Since the menu is positioned using calculations, I can't seem to use css positioning to achieve the affect I want:
![image](https://github.com/user-attachments/assets/b21e6aa5-a076-49f1-b0d3-43eea6ebe974)

This is what I have done so far:

![image](https://github.com/user-attachments/assets/ffbfa599-d3a8-473a-8a89-4ae13db7302e)

The arrow is hidden of course because of the positioning of the menu it self. Any ideas on how I can achieve this? Or which would be a better component to use?

I have created a pen as an example: https://codepen.io/jonathancounihan/pen/bGPXJVa

Thanks a bunch.

**Comment by rtcpw** [comment]
Quasar Portal is just a wrapper around Teleport:

https://github.com/quasarframework/quasar/blob/5cf9ad7eb340077d179194f9372e686f762775da/ui/src/composables/private.use-portal/use-portal.js

For some reason, this is private and internal to Quasar framework.
